File: Function.scala
def function(another_func: () => Unit) = {
  another_func()
}

def another_func() = println("Hello World!")

function(another_func)

def function1(another_func: => Unit) = {
  another_func
}

// function1(println("Hello World!"))
function1 { println("Hello World!") }


def function2(func: Int => Unit) = {
  for(i <- 1 to 3) func(i)
}

def helloNumber(i: Int) = {
  println("Hello " + i )
}

function2(helloNumber)

File: function.rb
def function(another_func)
  another_func.call
end

def another_func
  puts "Hello World!"
end

function(method(:another_func)) # prints "Hello World!"

def function(&block)
  block.call
end

function { puts "Hello World!" }

def iter(hproc)
  [1, 2, 3].each { |i| hproc.call(i) }
end

iter lambda { |i| puts "Hello #{i}" }
